src/boot/AppEventHandlers.js

```diff
@@ -89,6 +91,22 @@ class AppEventHandlers extends PureComponent<Props> {
     if (state === 'background' && Platform.OS === 'android') {
       NativeModules.BadgeCountUpdaterModule.setBadgeCount(unreadCount);
     }
+    if (state === 'active' && Platform.OS === 'ios') {
+      // TODO: This API is deprecated in react-native-notifications release
+      // 2.0.6-snapshot.8; see #3647.
+      try {
+        // We don't know the behavior if this is called before
+        // NotificationsIOS.requestPermissions(), so, catch any errors
+        // silently. Ray's investigation shows that it *shouldn't*
+        // throw, but may (https://github.com/zulip/zulip-mobile/pull/3947#discussion_r389192513).
+        NotificationsIOS.consumeBackgroundQueue();
+      } catch (e) {
+        logging.warn(e, {
+          message:
+            'Call to NotificationsIOS.consumeBackgroundQueue failed; pressed notification failed to navigate',
+        });
+      }
+    }
```

Comment:
Let's move all this into its own function, in our notifications code in `src/notifications/index.js` -- I think that's a better home for this complexity to live, next to the other code that has to deal with the intricacies of these libraries. Perhaps right after `handleInitialNotification`, which has a similar job.

Probably the `Platform.OS === 'ios'` condition belongs inside that function, too. Then this can just look like
```js
  if (state === 'active') {
    notificationOnAppActive();
  }
```
or some other name that indicates this is the function for doing notification-related things that we do when the app moves into the foreground / into state `active`. (And the details of what those things are is the job of the function's implementation, as part of the notification code.)

Reply:
OK, sounds good!